extensions.md

```diff
@@ -0,0 +1,15 @@
+# DSSE Extensions
+
+August 21, 2023
+
+Version 1.1.0-draft
+
+This document lists the well known DSSE
+[signature extensions](/envelope.md#signature-extensions).
+
+| Name (with link) | Kind | Notes |
+|------------------|------|-------|
+| [Sigstore] | `application/vnd.dev.sigstore.verificationmaterial;version=<VERSION>` | The X.509 certificate chain must not include the root or trusted intermediate certificates |
```

Comment:
@jku One thing we were discussing is a concern about recursion, where an extension might contain a DSSE envelope which could contain an extension, and so on. For this specific example for Sigstore, this won't be an issue because we're using [VerificationMaterial](https://github.com/sigstore/protobuf-specs/blob/d96fa4407cb4489233f133ab036c0ed98c2756a4/protos/sigstore_bundle.proto#L59) rather than the bundle. The former won't contain a DSSE, the latter would.

I think this is worth noting though, that extensions should (must?) not contain a DSSE to avoid recursion.

Thoughts @adityasaky?

Reply:
There's a statement about an extension not containing an envelope in envelope.md. Resolving this thread for now.